toolchain/check/eval.cpp

```diff
@@ -1727,13 +1727,15 @@ static auto TryEvalTypedInst(EvalContext& eval_context, SemIR::InstId inst_id,
       // Couldn't perform the action because it's still dependent.
       return MakeConstantResult(eval_context.context(), inst,
                                 Phase::TemplateSymbolic);
-    } else {
+    } else if constexpr (InstT::Kind.constant_needs_inst_id()) {
       return ConvertEvalResultToConstantId(
           eval_context.context(),
-          EvalConstantInst(eval_context.context(),
-                           eval_context.GetDiagnosticLoc({inst_id}),
-                           inst.As<InstT>()),
+          EvalConstantInst(eval_context.context(), inst_id, inst.As<InstT>()),
```

Comment:
Should we CHECK that the inst id has a value?

Reply:
Done.